core/src/main/java/org/mule/runtime/core/internal/event/DefaultEventBuilder.java

```diff
@@ -107,9 +107,13 @@ public DefaultEventBuilder(InternalEvent event) {
     this.session = event.getSession();
     this.error = event.getError().orElse(null);
     this.notificationsEnabled = event.isNotificationsEnabled();
-
     this.originalVars = (CaseInsensitiveHashMap<String, TypedValue<?>>) event.getVariables();
     this.loggingVariables = (CaseInsensitiveHashMap<String, String>) event.getLoggingVariables().orElse(null);
+    CaseInsensitiveHashMap<String, String> loggingVariablesCopyMap = new CaseInsensitiveHashMap<>();
```

Comment:
this instantiation is only needed in case it will be set (when loggingVariable is null)

Reply:
done!